app/frontend/src/Controller/ShabadInfo.js

```diff
@@ -39,7 +39,7 @@ const ShabadInfo = () => {
   const barIcon = isPopoverOpen ? faTimesCircle : faInfoCircle
 
   const onCopyClick = () => {
-    const gurmukhi = lines.reduce( ( text, { gurmukhi } ) => `${text}${gurmukhi}`, '' )
+    const gurmukhi = lines.reduce( ( text, { gurmukhi } ) => `${text}${gurmukhi} `, '' )
```

Comment:
Would this not cause an empty space at the end of the string?

Reply:
if he had done `' '` instead of `''`,  that would have caused an empty space

Reply:
wait no you're right there would be one extra at the end

Reply:
I tried that first but didn't work.
> if he had done `' '` instead of `''`, that would have caused an empty space

Reply:
> Would this not cause an empty space at the end of the string?

That is what we want. We are just making a long string by concatenating the lines.

Reply:
We do not want an exmpty space at the end of the string. Another way:

```suggestion
    const gurmukhi = lines.map( ( { gurmukhi } ) => gurmukhi ).join( ' ' )
```

This will only insert spaces in-between array elements